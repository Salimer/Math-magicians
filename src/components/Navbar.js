import { NavLink } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: '/calculate', text: 'Calculate' },
  { path: '/quote', text: 'Quote' },
];
const Navbar = () => (
  <nav className="navbar">
    <h1>Math Magicians</h1>
    <ul className="navigation">
      {links.map((link) => (
        <li key={link.text}>
          <NavLink to={link.path}>{link.text}</NavLink>
        </li>
      ))}
    </ul>
  </nav>
);
export default Navbar;
