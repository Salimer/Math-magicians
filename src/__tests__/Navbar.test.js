import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders Navbar correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
