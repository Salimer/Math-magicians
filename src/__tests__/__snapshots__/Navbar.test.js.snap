// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div>
  <nav
    class="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul
      class="navigation"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/calculate"
        >
          Calculate
        </a>
      </li>
      <li>
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
